lib/domain/usecase/send_media_on_web_with_caption_interactor.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:fluffychat/presentation/extensions/send_file_extension.dart';
+import 'package:fluffychat/presentation/extensions/send_file_web_extension.dart';
+import 'package:matrix/matrix.dart';
+
+class SendMediaOnWebWithCaptionInteractor {
+  Future<void> execute({
+    required Room room,
+    required MatrixFile media,
+    String caption = "",
+  }) async {
+    try {
+      final txid = room.client.generateUniqueTransactionId();
+      room.sendingFilePlaceholders[txid] = media;
+      final fakeImageEvent = await room.sendFakeImageEvent(
+        media,
+        txid: txid,
+      );
+      String? messageID;
+      Map<String, dynamic>? msgEventContent;
+      if (caption.isNotEmpty) {
+        messageID = room.client.generateUniqueTransactionId();
+        msgEventContent = room.getEventContentFromMsgText(message: caption);
+        await room.sendFakeMessage(
+          content: msgEventContent,
+          messageId: messageID,
+        );
+      }
+
+      await room.sendFileOnWebEvent(
+        media,
+        fakeImageEvent: fakeImageEvent,
+        txid: txid,
+      );
+      if (caption.isNotEmpty) {
+        await room.sendMessageContent(
+          EventTypes.Message,
+          msgEventContent!,
```

Comment:
What is happen if `msgEventContent` is null

Reply:
`caption.isNotEmpty` will guarantee that `msgEventContent != null` 
<img width="608" alt="Screenshot 2023-12-05 at 10 06 52" src="https://github.com/linagora/twake-on-matrix/assets/43041967/9550e51f-e6eb-4f96-8c18-78821d294596">

Reply:
oh shit, i think use `if (msgEventContent != null)`